### PR TITLE
Make ovnkube-trace to work for ovnkube-controller

### PR DESCRIPTION
**Pull request by pperiyasamy**
The ovnkube-node container is merged into ovnkube-controller container in an interconnect environment with this PR https://github.com/ovn-org/ovn-kubernetes/pull/3807, it creates regression in ovnkube-trace command. Hence this commit provides relevant fixes for the command so that it works seamlessly on this environment.

### CODE REVIEW

File: go-controller/cmd/ovnkube-trace/ovnkube-trace.go [hunk 1]
```diff
@@ -98,7 +100,8 @@ type PodInfo struct {
 	OfportNum            string // ofport number of veth interface or for host net pods of ovn-k8s-mp0
 	PodName              string // name of the pod
 	PodNamespace         string // the pod's namespace
-	ContainerName        string // the pod's principal container name (the first container found atm)
+	ContainerName        string // the pod's principal container name (the first container found atm
+	OvnKubeContainerName string // the pod's ovnkube node pod's container name.
```
Comment (kyrtapz) [inline go-controller/cmd/ovnkube-trace/ovnkube-trace.go:104]: NIT: that comment is a bit confusing, maybe change it to:
```
name of the container running ovnkube-node component
```

**Review by kyrtapz — APPROVED** [review]
Overall LGTM, small NITs that do not have to be addressed.

File: go-controller/cmd/ovnkube-trace/ovnkube-trace.go [hunk 2]
```diff
@@ -512,7 +515,7 @@ func getPodInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config,
 	// destination.
 	podInfo.OvnK8sMp0PortName = types.K8sMgmtIntfName
 	portCmd := fmt.Sprintf("ovs-vsctl get Interface %s ofport", podInfo.OvnK8sMp0PortName)
-	localOutput, localError, err := execInPod(coreclient, restconfig, ovnNamespace, podInfo.OvnKubePodName, "ovnkube-node", portCmd, "")
+	localOutput, localError, err := execInPod(coreclient, restconfig, ovnNamespace, podInfo.OvnKubePodName, podInfo.OvnKubeContainerName, portCmd, "")
```
[the remark this answers is not in the document]
Comment (pperiyasamy) [reply go-controller/cmd/ovnkube-trace/ovnkube-trace.go:518]: The podInfo containing actual pod object where tracing needs to be collected, but the command that's being run on its ovnk pod/container. so i think it's better to pass actual pod and container names to execInPod which seems to be more clear.

File: go-controller/cmd/ovnkube-trace/ovnkube-trace.go [hunk 3]
```diff
@@ -98,7 +100,8 @@ type PodInfo struct {
 	OfportNum            string // ofport number of veth interface or for host net pods of ovn-k8s-mp0
 	PodName              string // name of the pod
 	PodNamespace         string // the pod's namespace
-	ContainerName        string // the pod's principal container name (the first container found atm)
+	ContainerName        string // the pod's principal container name (the first container found atm
```
[the remark this answers is not in the document]
Comment (pperiyasamy) [reply go-controller/cmd/ovnkube-trace/ovnkube-trace.go]: oh! yes, done.

**Pull request merged**